src/include/duckdb/main/appender.hpp

```diff
@@ -130,6 +141,8 @@ class Appender : public BaseAppender {
 
 public:
 	void AppendDefault();
+	void SetColumn(const string &name) override;
+	void ResetColumn(const string &name) override;
```

Comment:
Instead of `SetColumn` and `ResetColumn`, perhaps we can have `AddColumn(...)` and `ClearColumns()`?